src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -552,9 +552,9 @@ public SolutionState AddProject(ProjectInfo projectInfo)
         }
 
         private static IEnumerable<TextDocumentState> GetDocumentStates(ProjectState projectState)
-            => projectState.DocumentStates.States
-                   .Concat(projectState.AdditionalDocumentStates.States)
-                   .Concat(projectState.AnalyzerConfigDocumentStates.States);
+            => projectState.DocumentStates.States.Values
```

Comment:
Doesn't values allocate anyway?

https://source.dot.net/#System.Collections.Immutable/System/Collections/Immutable/ImmutableSortedDictionary_2.Node.cs,b345c878e7fc2554

Reply:
➡️ This line is the same in this pull request as it was before this pull request. The performance improvement was applied targeting the single hot path method which is now marked with `[PerformanceSensitive]`. Other methods were optimized where trivial, or left the same.